app/package/controllers.py

```diff
@@ -82,8 +82,6 @@ def tag_data_package(publisher, package):
         status_db = Package.create_or_update_tag(publisher, package, data['version'])
         status_bitstore = bitstore.copy_to_new_version(data['version'])
 
-        if status_db is False or status_bitstore is False:
```

Comment:
We do want to know if these fail - this is quite a big deal. In fact, the logic here is not great. If status_db fails we don't want to continue and if status_bistore fails we want to roll back the previous creation of the tag and throw and exception.

Irrespective of this more general logic point, I would say we want the raise Exception here like w were doing.